### PR TITLE
feat: add queries for tss funds migrator info

### PR DESCRIPTION
**Pull request by kingpinXD**
# Description

Closes https://github.com/zeta-chain/node/issues/2371

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Include instructions and any relevant details so others can reproduce. 

- [ ] Tested CCTX in localnet
- [ ] Tested in development environment
- [ ] Go unit tests
- [ ] Go integration tests
- [ ] Tested via GitHub Actions 

# Checklist:

- [ ] I have added unit tests that prove my fix feature works

## Summary by CodeRabbit

- **New Features**
  - Introduced CLI commands `list-tss-funds-migrator` and `show-tss-funds-migrator` for querying TSS funds migrator information.
  - Added two new API endpoints for querying TSS fund migrator information.

- **Documentation**
  - Updated CLI documentation to include details for the new commands `list-tss-funds-migrator` and `show-tss-funds-migrator`.
  - Added new CLI documentation files detailing the usage and options for the new TSS funds migrator commands.

- **Tests**
  - Added test functions for retrieving and handling TSS funds migrator information.

- **Chores**
  - Minor parameter renaming for clarity in internal function definitions.

### CODE REVIEW

File: x/observer/client/cli/query_tss_fund_migrator.go [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+	"github.com/cosmos/cosmos-sdk/client"
+	"github.com/cosmos/cosmos-sdk/client/flags"
+	"github.com/spf13/cobra"
+
+	"github.com/zeta-chain/zetacore/x/observer/types"
+)
+
+func CmdGetTssFundsMigrator() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "show-tss-funds-migrator [chain-id]",
```
Comment (kingpinXD) [inline x/observer/client/cli/query_tss_fund_migrator.go:15]: Using `show` for single and `list` for multiple, as this seems to be the general terminology across the code. 
We can consider moving to `get` and `getAll`, which is a bit easier to understand

Comment (lumtis) [reply x/observer/client/cli/query_tss_fund_migrator.go:16]: Show sounds better to me

File: x/observer/client/cli/query.go [hunk 2]
```diff
@@ -41,6 +41,8 @@ func GetQueryCmd(_ string) *cobra.Command {
 		CmdListChainNonces(),
 		CmdShowChainNonces(),
 		CmdListPendingNonces(),
+		CmdGetAllTssFundsMigrator(),
+		CmdGetTssFundsMigrator(),
```
Comment (lumtis) [inline x/observer/client/cli/query.go:45]: When not using List and Show for the names here?

File: proto/zetachain/zetacore/observer/query.proto [hunk 3]
```diff
@@ -150,6 +151,30 @@ service Query {
...
+message QueryTssFundsMigratorInfoAllRequest {}
+
+message QueryTssFundsMigratorInfoAllResponse {
+  repeated TssFundMigratorInfo tss_funds_migrators = 1
+      [ (gogoproto.nullable) = false ];
+}
+
+message QueryTssFundsMigratorInfoRequest { int64 chain_id = 1; }
+message QueryTssFundsMigratorInfoResponse {
+  TssFundMigratorInfo tss_funds_migrator = 1 [ (gogoproto.nullable) = false ];
```
Comment (lumtis) [inline proto/zetachain/zetacore/observer/query.proto:177]: We should stick to camel case for now if other queries are using camel case
I may also think there is a problem with TypeScript client generation when using snake case (this is why originally Ignite templates was generating with camel case)

File: docs/cli/zetacored/zetacored_query_observer.md [hunk 4]
```diff
@@ -36,6 +36,7 @@ zetacored query observer [flags]
 * [zetacored query observer list-node-account](zetacored_query_observer_list-node-account.md)	 - list all NodeAccount
 * [zetacored query observer list-observer-set](zetacored_query_observer_list-observer-set.md)	 - Query observer set
 * [zetacored query observer list-pending-nonces](zetacored_query_observer_list-pending-nonces.md)	 - shows a chainNonces
+* [zetacored query observer list-tss-funds-migrator](zetacored_query_observer_list-tss-funds-migrator.md)	 - list all tss funds migrators
```
[the remark this answers is not in the document]
Comment (kingpinXD) [reply docs/cli/zetacored/zetacored_query_observer.md:39]: i tried to follow the existing nomenclature , for example this is a query which lists all node account 
https://github.com/zeta-chain/zeta-node/blob/cf6696eb451ad3f6d1e4695652ae925a69b3fe10/x/observer/client/cli/query_node_account.go#L15-L15

**Review by lumtis — COMMENTED** [review]
@kingpinXD ready for rereview? I may see https://github.com/zeta-chain/node/pull/2372#discussion_r1650595788 is not fixed

File: proto/zetachain/zetacore/observer/query.proto (hunk 3 above)
Comment (kingpinXD) [reply proto/zetachain/zetacore/observer/query.proto]: 3ce96875387aecd2302b74b2d6f9d2bce8e30e26

File: proto/zetachain/zetacore/observer/query.proto [hunk 5]
```diff
@@ -150,6 +151,30 @@ service Query {
...
```
Comment (lumtis) [inline proto/zetachain/zetacore/observer/query.proto:177]: We need to update camel case for these as well

Comment (kingpinXD) [reply proto/zetachain/zetacore/observer/query.proto:177]: I can change it but we have been using this format fields in a struct before, for example 

https://github.com/zeta-chain/zeta-node/blob/3ce96875387aecd2302b74b2d6f9d2bce8e30e26/proto/zetachain/zetacore/observer/query.proto#L206-L208

https://github.com/zeta-chain/zeta-node/blob/3ce96875387aecd2302b74b2d6f9d2bce8e30e26/proto/zetachain/zetacore/observer/query.proto#L221-L224

**Review by lumtis — APPROVED** [review]

**Review by swift1337 — APPROVED** [review]

**Pull request merged**